opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/bootstrap.py

```diff
@@ -0,0 +1,174 @@
+    """
+    pip_list = _pip_freeze()
+    for package in libraries[library]:
+        if "{}==".format(package).lower() in pip_list:
+            logger.info(
+                "Existing %s installation detected.  Uninstalling.", package
+            )
+            _pip_uninstall(package)
+    _pip_install(instrumentation)
+
+
+def _pip_freeze():
+    return (
+        subprocess.check_output([sys.executable, "-m", "pip", "freeze"])
+        .decode()
+        .lower()
+    )
+
+
+def _pip_install(package):
+    # explicit upgrade strategy to override potential pip config
+    subprocess.check_call(
+        [
+            sys.executable,
+            "-m",
+            "pip",
+            "install",
+            "-U",
+            "--upgrade-strategy",
+            "only-if-needed",
+            package,
+        ]
+    )
+
+
+def _pip_uninstall(package):
+    subprocess.check_call(
+        [sys.executable, "-m", "pip", "uninstall", "-y", package]
+    )
+
+
+def _pip_check():
+    """Ensures none of the instrumentations have dependency conflicts.
+    Clean check reported as:
+    'No broken requirements found.'
+    Dependency conflicts are reported as:
+    'opentelemetry-ext-flask 1.0.1 has requirement opentelemetry-sdk<2.0,>=1.0, but you have opentelemetry-sdk 0.5.'
+    To not be too restrictive, we'll only check for relevant packages.
+    """
+    check_pipe = subprocess.Popen(
+        [sys.executable, "-m", "pip", "check"], stdout=subprocess.PIPE
+    )
+    pip_check = check_pipe.communicate()[0].decode()
+    pip_check_lower = pip_check.lower()
+    for package_tup in libraries.values():
+        for package in package_tup:
+            if package.lower() in pip_check_lower:
+                raise RuntimeError(
+                    "Dependency conflict found: {}".format(pip_check)
+                )
+
+
+def _is_installed(library):
+    return library in sys.modules or pkgutil.find_loader(library) is not None
+
+
+def _find_installed_libraries():
+    return {k: v for k, v in instrumentations.items() if _is_installed(k)}
+
+
+def _run_requirments(packages):
+    print("\n".join(packages.values()), end="")
+
+
+def _run_install(packages):
+    for pkg, inst in packages.items():
+        _install_package(pkg, inst)
+
+    _pip_check()
+
+
+def run() -> None:
+    parser = argparse.ArgumentParser(
+        description="""
+        opentelemetry-bootstrap detects installed libraries and automatically installs the relevant
+        instrumentation packages for them.
+        """
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        choices=[_OUTPUT_INSTALL, _OUTPUT_REQUIREMENTS],
+        default=_OUTPUT_INSTALL,
+        help="""
+        install - uses pip to install the new requirements using to the currently active site-package.
+        requirements - prints out the new requirements to stdout. Output can be piped and appended to
+                       a requirements.txt file.
+        """,
+    )
```

Comment:
Do we really need to have the intermediate `--output` command? 
`opentelemetry-bootstrap -o install` looks weird to me.

What about removing it and having something like the following?
```
opentelemetry-bootstrap --install
opentelemetry-bootstrap --requirements
```
Btw, should a call without arguments (`opentelemetry-bootstrap`) perform the installation? I got surprised by it, I run `opentelemetry-bootstrap` to see what was it about and suddenly it was performing a lot of installations.

Reply:
I assumed we might have a more actions in future like updating requirements.txt file in place or support for other tools like pipenv, poetry, etc. Would `--action/-a` be a better name for the flag instead of `--output/-o` ?

Reply:
Replaced with `--action/-a` instead and change the default to printing requirements.